dev/reloadable-server/src/main/scala/play/core/server/LagomReloadableDevServerStart.scala

```diff
@@ -133,9 +133,14 @@ object LagomReloadableDevServerStart {
               // Let's load the application on another thread
               // as we are now on the Netty IO thread.
               //
-              // Because we are on DEV mode here, it doesn't really matter
-              // but it's more coherent with the way it works in PROD mode.
-              implicit val ec = play.core.Execution.internalContext
+              // this whole Await.result(Future{}) thing has been revisited in Play
+              // but the issue is not yet fully addressed there neither
+              // see issues:
+              // https://github.com/playframework/playframework/pull/7627
+              // and https://github.com/playframework/playframework/pull/7644
+              //
+              // we should reconsider if still need to have our own reloadable server
+              implicit val ec = ExecutionContext.global
```

Comment:
This code was originally picked from Play and in the meantime, it involved quite a lot there. 

The current situation is that, in Play, we simply block on the current thread and we have a `synchronized` block around it and there is TODO to improve it further. 

Ideally, we should revisit both code bases. Streamline it in Play and let Lagom reuses Play code.